third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -492,7 +504,18 @@ def arcs_kt_jvm_test_suite(
         "medium", "large".
       flaky: boolean indicating whether the test is flaky and should be re-run
         on failure.
+      build_flag: Optional name of an Arcs build flag that conditionally guards this test rule.
```

Comment:
This is probably not a good idea (right now). We still want to build and test features that have status NOT_READY, otherwise it's too easy for them to become out of date. 

It's particularly difficult here because the code generator's output is changing, and I'm assuming that the test won't actually compile when the flag is disabled?

I think what we'll probably need here is a way of invoking the code generator with a particular set of overridden flags, e.g. something like this:

```
arcs_kt_gen(
    name = "codegen_with_type_slicing",
    flag_overrides = {"particle_type_slicing": True},
    srcs = [
        "type_slicing.arcs",
    ],
)
```

then you can add a dep on that to your test suite to know that it will always test the code with the flag correctly set.

Reply:
Yes, the problem is that the non-generated test code only compiles if the full slicing logic is being produced by codegen. As far as I can tell there's no way to include `TypeSlicingTest.kt` without the codegen change enabled.

Reply:
To clarify:
```
// particle Foo
//   data: writes {...}

// with the flag off:
typealias Foo_Data = AbstractFoo.Foo_Data
typealias Foo_Data_Slice = AbstractFoo.Foo_Data
abstract class AbstractFoo {
  class Foo_Data : EntityBase() { ... }
}

// with the flag on:
typealias Foo_Data = AbstractFoo.Foo_Data
typealias Foo_Data_Slice = AbstractFoo.Foo_Data_Slice
abstract class AbstractFoo {
  interface Foo_Data_Slice : Entity { ... }
  class Foo_Data : EntityBase(), Foo_Data_Slice { ... }
}
```
So even if we had codegen produce `Foo_Data_Slice` but not connect `Foo_Data` to it and set the unit test up to ignore the typealiases and reference the interface and classes directly, it still won't compile because it requires `Foo_Data` to implicitly convert to its slice form (and then up to other compatible entity slices in the lattice).

Reply:
What I meant was:

Currently you have this build rule for your test:

```
arcs_kt_jvm_test_suite(
    name = "type_slicing",
    srcs = ["TypeSlicingTest.kt"],
    build_flag = "particle_type_slicing",
    deps = [":codegen", ...],
)

arcs_kt_gen(
    name = "codegen",
    srcs = ["type_slicing.arcs"],
)
```

where `build_flag = "particle_type_slicing"` indicates that the whole target should be skipped if the flag is not enabled.

What I'm suggesting instead is something more like this:

```
arcs_kt_jvm_test_suite(
    name = "type_slicing",
    srcs = ["TypeSlicingTest.kt"],
    deps = [":codegen", ...],
)

arcs_kt_gen(
    name = "codegen",
    srcs = ["type_slicing.arcs"],
    force_enable_type_slicing = True,
)
```

Instead of using the default value of the flag, you force type slicing to be enabled _just for that one specific `codegen` target_. That way, the code generated by `codegen` will always have type slicing, and so you'll always be able to build and run the `type_slicing` test.

You'll have to plumb that `force_enable_type_slicing` param through to the schema2wasm invocation, like you've already done.